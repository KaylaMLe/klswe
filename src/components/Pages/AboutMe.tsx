import { css } from '@emotion/css';
import React from 'react';
import placeholder from '../../assets/images/portrait-placeholder.png';

export function AboutMe(): React.JSX.Element {
	const textBoxStyle = css({
		backgroundColor: '#00007A',
		color: '#CCCCFF',
		boxShadow: '0 0 5px 5px #00007A',
		fontSize: '12pt',
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
		justifyContent: 'center',
		boxSizing: 'border-box',
		padding: '1rem',
		height: '90vh',
	});

	return (
		<div className={textBoxStyle}>
			<Portrait />
			<p>
				I'm a passionate web developer on the brink of graduation, eager to dive into the tech world
				with fresh ideas. This website is my digital canvas, meticulously crafted to showcase my
				technical skills, creativity, and the projects I'm proud of. My journey in web development
				blends analytical thinking with aesthetic design, aiming to create seamless, user-friendly
				experiences. As I stand at the threshold of my professional career, I'm enthusiastic about
				securing my first full-time software engineering role, eager to contribute, learn, and grow
				within a dynamic team.
			</p>
		</div>
	);
}

function Portrait(): React.JSX.Element {
	const portraitStyle = css({
		borderRadius: '50%',
		height: '192px',
		width: '192px',
	});

	const imgSrc = process.env.REACT_APP_PORTRAIT_IMG_URL || placeholder;

	return (
		<img
			id='main'
			className={portraitStyle}
			src={imgSrc}
			alt='A professional head shot of myself'
		/>
	);
}