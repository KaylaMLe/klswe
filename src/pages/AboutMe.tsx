import { css } from "@emotion/css";
import { ReactElement } from "react";
import { NavBar } from "../NavBar";

export function AboutMe(): ReactElement {
	return (
		<div>
			<NavBar />
			<TextBox />
		</div>
	);
}

function TextBox(): ReactElement {
	const textBoxStyle = css({
		color: '#A3A3FF',
		borderRadius: '0.1rem',
		backgroundColor: '#00007A',
		boxShadow: '0 0 5px 5px #00007A',
		marginTop: '4px',
		marginLeft: '2px',
		marginRight: '2px',
		padding: '1rem',
	});

	return <div className={textBoxStyle}>Hi!</div>;
}