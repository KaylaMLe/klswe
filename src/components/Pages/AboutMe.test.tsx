import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import AboutMe from './AboutMe';

jest.mock('../../constants', () => ({
	__esModule: true,
	ABOUT_ME_TXT: 'Hi!',
	PORTRAIT_IMG_URL: '<rootdir>/src/assets/images/portrait-placeholder.jpg',
}));

describe('AboutMe', () => {
	beforeEach(() => {
		render(
			<BrowserRouter>
				<AboutMe />
			</BrowserRouter>
		);
	});

	it('renders the portrait', () => {
		const imgElement = screen.getByAltText(/Professional head shot of myself/);
		expect(imgElement).toBeInTheDocument();
	});

	it('has a highlighted navigation button', () => {
		const navBtn = screen.getByText(/About me/);
		expect(navBtn).toHaveStyle('background-color: #11D45C');
		expect(navBtn).toHaveStyle('color: #16161D');
	});
});