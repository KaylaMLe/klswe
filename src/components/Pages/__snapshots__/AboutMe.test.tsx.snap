// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe renders properly on different display sizes 1`] = `
<div
  style="width: 1024px; height: 768px;"
>
  <div
    class="css-ccy2vc"
  >
    <nav
      class="css-zw4cqh"
    >
      <a
        class="css-42nl7f"
        href="#main"
      >
        Skip to main content
      </a>
      <div
        class="css-1iarqv4"
      >
        <a
          class="css-1x0xomp"
          href="/"
        >
          <h1
            class="css-dj9cgy"
          >
            Kayla Le
          </h1>
        </a>
        <h2
          class="css-1b38hje"
        >
          Software Engineer
        </h2>
      </div>
      <div
        class="css-1p063y5"
      >
        <a
          class="css-23kpuw"
          href="/about-me"
        >
          About me
        </a>
        <nav
          class="css-z6jdx8"
        >
          <button
            aria-controls="dropdown1"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="css-4r3yzx"
          >
            Projects
            <div
              class="css-144hx5m"
              role="none"
            >
              ▼
            </div>
          </button>
        </nav>
      </div>
    </nav>
    <div
      class="css-v2kfba"
      id="main"
      role="none"
      tabindex="-1"
    >
      <div
        class="css-1qe9sr2"
      >
        <img
          alt="Professional head shot of myself"
          class="css-cwyjqr"
          src="test-file-stub"
        />
        <p
          class="css-ocylvi"
        >
          Hi!
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`AboutMe renders properly on different display sizes 2`] = `
<div
  style="width: 375px; height: 812px;"
>
  <div
    class="css-ccy2vc"
  >
    <nav
      class="css-zw4cqh"
    >
      <a
        class="css-42nl7f"
        href="#main"
      >
        Skip to main content
      </a>
      <div
        class="css-1iarqv4"
      >
        <a
          class="css-1x0xomp"
          href="/"
        >
          <h1
            class="css-dj9cgy"
          >
            Kayla Le
          </h1>
        </a>
        <h2
          class="css-1b38hje"
        >
          Software Engineer
        </h2>
      </div>
      <div
        class="css-1p063y5"
      >
        <a
          class="css-23kpuw"
          href="/about-me"
        >
          About me
        </a>
        <nav
          class="css-z6jdx8"
        >
          <button
            aria-controls="dropdown1"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="css-4r3yzx"
          >
            Projects
            <div
              class="css-144hx5m"
              role="none"
            >
              ▼
            </div>
          </button>
        </nav>
      </div>
    </nav>
    <div
      class="css-v2kfba"
      id="main"
      role="none"
      tabindex="-1"
    >
      <div
        class="css-1qe9sr2"
      >
        <img
          alt="Professional head shot of myself"
          class="css-cwyjqr"
          src="test-file-stub"
        />
        <p
          class="css-ocylvi"
        >
          Hi!
        </p>
      </div>
    </div>
  </div>
</div>
`;
